Source/Scene/Model.js

```diff
@@ -1209,9 +1211,14 @@ define([
         var basePath = defaultValue(options.basePath, modelResource.clone());
         var resource = Resource.createIfNeeded(basePath);
 
-        // If no cache key is provided, use the absolute URL, since two URLs with
-        // different relative paths could point to the same model.
-        var cacheKey = defaultValue(options.cacheKey, getAbsoluteUri(modelResource.url));
+        // If no cache key is provided, use a GUID.
+        // Check using a URI to GUID dictionary that we have not already added this model.
+        var cacheKey = defaultValue(options.cacheKey, defaultValue(uriToGuid[getAbsoluteUri(modelResource.url)], createGuid()));
```

Comment:
Isn't this still going to be problematic for data uris because we're using that as the key in the dictionary?

Reply:
Excellent question! So the uriToGuid's purpose is to map the absolute URL to GUID to prevent adding the same model twice to the cache. This map is internal to Model.js, and thus duplication of data is minimal (as I'm new to Javascript any suggestions for improvement are welcome). The memory issue arises elsewhere when the model is cached in the animation system. Specifically, on line 51 in ModelAnimationCache.js the model's cacheKey will be used repeatedly to form a composite animation cache key. By using the GUID we avoid using the data URI here and thus prevent the system from quickly exhausting memory.